twoliter/embedded/rpm2img

```diff
@@ -159,8 +159,8 @@ rpm -iv --ignorearch --root "${ROOT_MOUNT}" "${PACKAGE_DIR}"/*.rpm
 # Inventory installed packages.
 INVENTORY_QUERY="\{\"Name\":\"%{NAME}\"\
 ,\"Publisher\":\"Bottlerocket\"\
-,\"Version\":\"${VERSION_ID}\"\
-,\"Release\":\"${BUILD_ID}\"\
+,\"Version\":\"%{Version}\"\
```

Comment:
Let's split out the change to `rpm2img` as a separate commit, since this is where the actual externally visible change happens.
```suggestion
,\"Epoch\":\"1\"\
,\"Version\":\"%{Version}\"\
```

Reply:
> Let's split out the change to `rpm2img` as a separate commit, since this is where the actual externally visible change happens.

What is the "Epoch" in this suggestion? When would it change?

Also, as usual, shell is a mystery to me and I don't understand what `%{Version}` is? Where is that defined?

Currently the only documentation of this json output is `# Inventory installed packages.`. Should we update that to say more about the fields and what they mean?

Reply:
"Epoch" is essentially a trump card to version comparisons. For example, a Bottlerocket 1.20.0 variants includes runc 1.1.12, but under the old process of building application inventory, we specify the `Version` as Bottelerocket's release version - 1.20.0.  Going forward, we will specify package `Version` as the package's actual version, not the project (previously Bottlerocket monorepo)'s version - a variant would then specify, say runc 1.1.13, as `Version` 1.1.13. This is unfortunately lower than our `Version` in previous variants and would break inventory / security advisory comparisons. To circumvent this, packages in kits will start from `Epoch: "1"` (configurable to be overriden by `BUILDSYS_VERSION_BUILD_EPOCH`)


> Also, as usual, shell is a mystery to me and I don't understand what %{Version} is? Where is that defined?

This is querying RPM inventory. `Version` is the corresponding field in a package spec.

I can update the comment here to be more descriptive